core-bundle/src/Resources/contao/models/FilesModel.php

```diff
@@ -394,6 +395,44 @@ public static function findMultipleFoldersByFolder($strPath, array $arrOptions=a
 	protected function postSave($intType)
 	{
 	}
+
+	/**
+	 * Get the meta fields defined in `tl_files.meta.eval.metaFields`.
+	 */
+	public static function getMetaFields(): array
+	{
+		Controller::loadDataContainer('tl_files');
+
+		return array_keys($GLOBALS['TL_DCA']['tl_files']['fields']['meta']['eval']['metaFields'] ?? array());
+	}
+
+	/**
+	 * Get the meta data for this file. Specify one or more locales - meta
+	 * data of the first matching one will be returned or null if none was
+	 * found.
+	 */
+	public function getMetaData(string ...$locales): ?MetaData
+	{
+		$dataCollection = StringUtil::deserialize($this->meta, true);
+
+		foreach ($locales as $locale)
+		{
+			$data = $dataCollection[$locale] ?? array();
+
+			if (!empty($data))
+			{
+				// Make sure we resolve insert tags pointing to files.
+				if (isset($data[MetaData::VALUE_URL]))
+				{
+					$data[MetaData::VALUE_URL] = Controller::replaceInsertTags($data[MetaData::VALUE_URL]);
```

Comment:
> This is only necessary if the target URL would be a picture that's being generated, right? For example, if the caption is an insert tag to a page URL, it would simply be resolved when generating the page?

> In what case is the files path relevant, and would the insert tag result be a UUID or a path?
You get a path - e.g. when picking an other image `{{file::a-uuid}}`. I don't think we need to support UUIDs directly.